iterators_task_2/student/include/impl.h

```diff
@@ -1,21 +1,73 @@
 #pragma once
 #include <iterator>
+#include <functional>
 
 template <class Container>
 struct accumulator
 {
-    /** @todo Iterator traits*/
-    /** @todo Constructor with container*/
-    /** @todo lvalue writing operator*/
-    /** @todo rvalue writing operator*/
+    using value_type = typename Container::iterator::value_type;
```

Comment:
Does it have sense for output iterators?

Reply:
Done